snuba/datasets/errors_replacer.py

```diff
@@ -1159,16 +1159,23 @@ def process_delete_tag(
     tag_column_name = tag_column_map["tags"].get(tag, tag)
     is_promoted = tag in promoted_tags["tags"]
 
-    where = """\
-        WHERE project_id = %(project_id)s
-        AND received <= CAST('%(timestamp)s' AS DateTime)
+    # We cannot put the tag condition (which is what we are mutating) in the
+    # prewhere clause. This is because the prewhere clause is processed before
+    # the FINAL clause.
+    # So if we are trying to mutate a row that was mutated before but not merged
+    # yet, the PREWHERE would return the old row that has already been
+    # replaced.
+    prewhere = " PREWHERE project_id = %(project_id)s"
```

Comment:
I am guessing this is just for optimizing the query and not necessarily a requirement.

Reply:
Actually thinking over it, this is useless as the project_id is already in the primary key